File: client/src/lib/invoke/loadConfigFromFile.ts
import { invoke } from "@tauri-apps/api/core";
import { ResultAsync } from "neverthrow";
import type { Config } from "@bindings";

export function loadConfigFromFile(path_string: string): ResultAsync<Config, string> {
    return ResultAsync.fromPromise(
        invoke("load_config_from_file", { path_string }),
        (error) => `Failed to load config from file: ${error}`
    );
}

File: client/src-tauri/src/commands/load_config_from_file.rs
use crate::core::config::Config;

#[tauri::command]
pub async fn load_config_from_file(
    path_string: String,
) -> Result<Config, String> {
    tauri::async_runtime::spawn_blocking(|| Config::from_file(path_string))
        .await
        .map_err(|err| format!("Failed to load config from file: {err}"))?
}
